proof/access-control/ARM/ArchTypes.thy

```diff
@@ -0,0 +1,17 @@
+(*
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ *)
+
+theory ArchTypes
+imports Deterministic_AC
+begin
+
+section \<open>Policy and policy refinement\<close>
+
+subsection \<open>Arch-specific definitions\<close>
+
+datatype arch_auth = ASIDPoolMapsASID
```

Comment:
We discussed this a little, and I think we should keep ASIDs arch specific for now. We have them on all architectures that are currently supported, and I think eventually we should make them properly generic (in code as well as spec), but doing it just for access control is strange. So I'd like to leave that for later.

I'm also trying not to overload this too much, because this can easily turn into a half-year project, which is also why I was encouraging Ryan to make a PR early (only Access is really arch-split currently, but that is an important first milestone, and things would of course still change over time).

If you and Raf prefer to see the whole thing in one go, I'm fine with that too. It'll be large, but more complete.

Reply:
To clarify and update what I said earlier, I think it would be good to see this work in chunks, but I'd prefer it if each chunk contained a complete treatment of a small number of theories rather than an incomplete treatment of many theories. To confirm that a given subset of theories' arch-split is "complete", I would want to see:

1. A PR description clearly listing which theories have been arch split.
2. A theory should only be on the list if all the theories it imports were previously arch split, or are also on the list.
3. Theories not on the list should ideally only have minimal changes to adjust to the arch split in the listed theories.
4. For theories on the list which are generic:
  a. No wholesale `interpretation Arch`, except in context blocks whose sole purpose is to explicitly requalify some names. SImilarly, no `context Arch` or explicit qualified references to names in the `Arch` context.
  b. Requalified names should be expected to exist on all architectures, and have the same "shape" on all architectures.
5. For theories on the list which are in an arch subdirectory:
  a. Most definitions and theorems should be in a `context Arch` block with a `global_naming` declaration named after the current architecture. The main exception is for `global_interpretation`s for any uses of the `locale` trick.
  b. The general aim is to minimise the amount of stuff in arch theories without breaking 3b.

The combination of 2 and 4a is particularly important, since without those, it's difficult to tell whether anything else makes sense. That's the main reason this PR (at the time I first looked at it) didn't meet these criteria: `Access.thy` is the first theory after AInvs that needs arch split work, and it still had a large context block with an `interpretation Arch`.

I see that there have been some updates since I first looked, and you've commented on another thread that the intention was to just arch-split `Access.thy`, so I'll take another look soon... Also, it would be good to know the context/goals for this work.

Reply:
Very happy with that process.